src/lib.rs

```diff
@@ -120,9 +120,13 @@ use regex::Regex;
 /// # Ok(())
 /// # }
 /// ```
-pub fn from_str<'a, T>(input: &'a str, regex: &str) -> std::result::Result<T, Error> where T: Deserialize<'a> {
-    let regex = Regex::new(&regex).map_err(Error::BadRegex)?;
-    from_str_regex(input, regex)
+#[inline]
```

Comment:
Do you have any benchmarks that show noticeable performance improvements marking this and other functions as inline?